api/helpers/mock-casual-match.js

```diff
@@ -0,0 +1,40 @@
+async function getRematchGames (game, priorRematchGames = []) {
+  const rematchGames = [game, ...priorRematchGames];
+  const gameToAdd = await Game.findOne({ rematchGame: game.id });
+  if (!gameToAdd) {
+    return rematchGames;
+  }
+  if (!gameToAdd.rematchGame) {
+    return rematchGames;
+  }
```

Comment:
Yeah I can't see how those wouldn't be equivalent. More concise; definitely better

Reply:
Actually it turns out we don't care if the new game has a rematchGame attribute because we're recursing _backward_ through time. We care if there is a game that _this game_ is a rematch _of._ So we only need the `if(!gameToAdd)` without checking `rematchGame` at all